fixbackend/notification/notification_router.py

```diff
@@ -28,12 +30,19 @@
 from fixbackend.logging_context import set_workspace_id, set_context
 from fixbackend.notification.model import WorkspaceAlert, AlertingSetting
 from fixbackend.notification.notification_service import NotificationService
+import jwt
 
 log = logging.getLogger(__name__)
 AddSlack = "notification_add_slack"
 AddDiscord = "notification_add_discord"
 
-State = "add-notification-channel"
+
+STATE_TOKEN_AUDIENCE = "fix:notification-state"
+
+
+def generate_state_token(data: Dict[str, str], secret: str) -> str:
```

Comment:
Consider defining the method inside `notification_router` where you have access to config.secret.